src/streams/openssl.c

```diff
@@ -737,10 +734,29 @@ int git_openssl_stream_new(git_stream **out, const char *host, const char *port)
 
 	*out = (git_stream *) st;
 	return 0;
+}
 
-out_err:
-	git_stream_free(st->io);
-	git__free(st);
+int git_openssl_stream_wrap(git_stream **out, git_stream *in, const char *host)
+{
+	return openssl_stream_wrap(out, in, host, 0);
+}
+
+int git_openssl_stream_new(git_stream **out, const char *host, const char *port)
+{
+	git_stream *stream = NULL;
+	int error;
+
+	assert(out && host);
```

Comment:
```suggestion
	assert(out && host && port);
```

Reply:
That's more restrictive than non-TLS sockets, which allow a `NULL` port, since `getaddrinfo` allows a `NULL` service as well.  I think that asserting that `port` be non-`NULL` makes sense, but we should think through the differences and make sure that we understand how it works (or - more likely - doesn't work) with a `NULL` port  today.